Doc/howto/logging-cookbook.rst

```diff
@@ -589,18 +589,16 @@ An example of using these two classes follows (imports omitted)::
     que = queue.Queue(-1)  # no limit on size
     queue_handler = QueueHandler(que)
     handler = logging.StreamHandler()
-    listener = QueueListener(que, handler)
     root = logging.getLogger()
     root.addHandler(queue_handler)
     formatter = logging.Formatter('%(threadName)s: %(message)s')
     handler.setFormatter(formatter)
-    listener.start()
-    # The log output will display the thread which generated
-    # the event (the main thread) rather than the internal
-    # thread which monitors the internal queue. This is what
-    # you want to happen.
-    root.warning('Look out!')
-    listener.stop()
+    with QueueListener(que, handler) as listener:
```

Comment:
Instead of this new usage replacing the older, I would like the old usage to be kept in the documentation, with the next paragraph showing the new usage close to the Python `versionchanged` which clearly shows that this way of doing things is restricted to particular Python versions.

Reply:
Updated.